File: Snippet/app/src/main/java/com/dvidal/snippet/infrastructure/MyApplication.java
package com.dvidal.snippet.infrastructure;

import android.app.Application;

import com.dvidal.snippet.di.AppComponent;
import com.dvidal.snippet.di.AppModule;
import com.dvidal.snippet.di.DaggerAppComponent;
import com.facebook.stetho.Stetho;
import com.uphyca.stetho_realm.RealmInspectorModulesProvider;
import io.realm.Realm;


/**
 * Created by diegovidal on 10/01/2018.
 */

public class MyApplication extends Application {

    private AppComponent component;

    @Override
    public void onCreate() {
        super.onCreate();
        Realm.init(this);

        Stetho.initialize(
                Stetho.newInitializerBuilder(this)
                        .enableDumpapp(
                                Stetho.defaultDumperPluginsProvider(this)
                        )
                        .enableWebKitInspector(
                                RealmInspectorModulesProvider
                                        .builder(this)
                                        .build()
                        )
                        .build());

        //Dagger
        component = DaggerAppComponent.builder().appModule(new AppModule(this)).build();
        component.inject(this);
    }

    public AppComponent getAppComponent() {
        return component;
    }
}

File: FindTheDay/app/src/main/java/br/com/vp/findtheday/MyApplication.kt
package br.com.vp.findtheday

import android.app.Application
import br.com.vp.findtheday.di.DaggerMyApplicationComponent
import br.com.vp.findtheday.di.MyApplicationComponent

/**
 * @author diegovidal on 05/06/2018.
 */
class MyApplication: Application() {

    lateinit var myApplicationComponent: MyApplicationComponent

    override fun onCreate() {
        super.onCreate()

        sharedInstance = this
        myApplicationComponent = DaggerMyApplicationComponent
                .builder().build()
    }

    companion object {

        lateinit var sharedInstance: MyApplication

        fun getComponent(): MyApplicationComponent {
            return sharedInstance.myApplicationComponent
        }
    }
}